implementations/rust/ockam/ockam_api/src/kafka/secure_channel_map.rs

```diff
@@ -0,0 +1,387 @@
+use crate::hop::Hop;
+use crate::kafka::{
+    KAFKA_SECURE_CHANNEL_CONTROLLER_ADDRESS, KAFKA_SECURE_CHANNEL_LISTENER_ADDRESS,
+    ORCHESTRATOR_KAFKA_CONSUMERS,
+};
+use ockam::remote::RemoteForwarder;
+use ockam_core::compat::collections::{HashMap, HashSet};
+use ockam_core::compat::sync::Arc;
+use ockam_core::errcode::{Kind, Origin};
+use ockam_core::Message;
+use ockam_core::{async_trait, route, Address, AllowAll, Error, Route, Routed, Worker};
+use ockam_identity::api::{
+    DecryptionRequest, DecryptionResponse, EncryptionRequest, EncryptionResponse,
+};
+use ockam_identity::authenticated_storage::AuthenticatedStorage;
+use ockam_identity::{Identity, IdentityVault, SecureChannelRegistryEntry, TrustEveryonePolicy};
+use ockam_node::compat::tokio::sync::Mutex;
+use ockam_node::Context;
+use serde::{Deserialize, Serialize};
+
+///Offer simple APIs to encrypt and decrypt kafka messages.
+/// Underneath it create secure channels for each topic/partition
+/// and uses them to encrypt the content.
+/// Multiple secure channels may be created for the same topic/partition
+/// but each will be explicitly label.
+/// It's the same for both producer and consumer although it could be split
+/// into two distinct implementation.
+///This is proxy trait to avoid propagating the vault implementation.
+#[async_trait]
+pub(crate) trait KafkaSecureChannelController: Send + Sync {
```

Comment:
Food for thought: how much of this is really Kafka-specific? It seems to me that the main idea is to allow the creation and tracking of secure channels and forwarders for a specific set of keys (in the HashMap sense). It turns out that those keys are made of topic names and partition ids for the Kafka use case but this could possibly be applicable to other use cases.